### DISCUSSION TITLE
Motion Snooze

### DISCUSSION THREAD

**Discussion by JB09** [tsightler/ring-mqtt#425] [Feature Requests]
It would be nice to be able to snooze motion notifications (globally or at device level).  For example, when I'm home I would like motion events to still record, I just don't want to keep receiving the alerts on my phone.  This could be accomplished by creating a switch that enables/disables a Device Schedule (Motion Schedule) from 12am-12:59pm.  It looks like their is some variables related to motion_snooze and schedule_settings in the data dump.  Maybe this is possible?

**Comment by tsightler** [comment]
It could also be accomplished by using HA for notifications instead of the Ring app, which is what a lot of people seem to do, since that gives you complete control of notifications however you want.  I believe it should also be possible by using the modes function, which is already supported in ring-mqtt, since it allows you to control notifications for different cameras based on the set mode (Home mode vs Away Mode for example), although there is a current report that this isn't working right now for some unknown reason (it's been supported for over 2 years, so not sure why it would be suddenly not working, but who knows, it's not a feature that I actually use).

Unfortunately, toggling these other advanced motion options are not currently available via native ring-client-api functions, and, while it's completely possible to call custom functions, someone has to do the API exploration/reverse engineering work to figure out the specific API calls required and it's not something I have the time to dig into at this point.  If someone else wants to dig into these options to determine the required API calls, I'm certainly open to adding these types of functions.

**Comment by JB09** [comment]
I tried working with modes, but that seems to toggle “motion detection” not “motion alerts”.  But maybe I am misunderstanding the setting or it has something to do with the current problem?

**Reply by tsightler** [reply]
No, the problem that was reported was that modes panel wasn't available in HA (at least, that's how it was described in the discussion thread about this).  It sounds like it worked fine for you and you are probably correct that modes disables detection a not alerts.

One issue with disabling alerts is that, if alerts are disabled for a camera, this will also stop any motion events being detected via ring-mqtt as well and most people don't seem to want that, which is why they end up configuring alerts from HA, because they still want HA to get all events.  However, I would be interested in adding the ability to toggle these alert options via ring-mqtt, but until I have a lot more time available, or someone else takes the time to dig into the API and determine what's needed, then it's not going to happen.

**Comment by JB09** [comment]
Maybe here: https://github.com/tchellomello/python-ring-doorbell/issues/126

**Reply by tsightler** [reply]
Looks promising, I can't really think of a practical way to implement motion snooze schedules that would work logically via MQTT or in Home Assistant though.  But the motion alerts and motion detection options seem pretty doable.  It still likely won't be any time soon as I don't have any time to dedicate to this project right now.

**Comment by JB09** [comment]
Completely understand.  Hopefully it can go on the roadmap!  My suggestion would be to toggle a schedule named “24-7” with a 12:00am-12:59pm or have the user make their own schedule in the app and then allow the user to define an attribute with the name to be published via mqtt as part of the toggle.  I’m happy to test whenever you get around to it, thanks for the quick responses!

**Comment by JB09** [comment]
Or if users make their own schedules in app, you pull the ruleId list for each camera at startup and then have HA create a binary_sensor for each ruleId

**Comment by tsightler** [comment]
Yeah, I'm not too crazy about either of those approaches honestly.  Creating a schedule feels too intrusive as I wouldn't like the behavior if I ran some script from the internet one time and it just made a bunch of schedules for each of my cameras, even if they were turned off by default, but perhaps a more palatable option is to create the schedule automatically only once the feature is ever toggled on the first time.

Using the existing schedules seems OK, but they can be created and removed at any time which means I'd really need to enhance the dynamic nature of the script, something I've often wanted to do, but haven't really had time to do to this point (the framework is there, and it wouldn't be THAT hard, but it's the testing of all of the cases that really scares me).  It would also be difficult to maintain sync between HA and the Ring app because, for example, if ring-mqtt wasn't running at the time, and a schedule was removed using the Ring app, this schedule wouldn't be discovered by ring-mqtt during the next run but it would just hang around as a ghost in HA since it was previously discovered.  Of course I could store such schedules in the state file and use that to clean up, but it's just more complication.

I'm not sure what the right answer is here, maybe I can come to terms with the idea of automatically creating a snooze schedule, but my initial thought was that, if a user wants this specific feature, they would have to manually create a schedule with a specific name (perhaps "Automated Snooze" or something like that) and, if ring-mqtt found that specific schedule for a camera, it would create the binary sensor.

**Reply by JB09** [reply]
The simplest option is probably to dictate what the user should name their motion schedule and then publish the entity to HA if it’s found at ring-mqtt boot.  If the user deletes the schedule, they’ll have a zombie entity but at least you’re limiting it to only 1 possible per camera.  And I would imagine most of your users have enough knowledge to manually delete an entity from HA if it came to that.

If at some future point you enhance the dynamic nature of the script the process could be improved to handle more customization/edge cases, but at least the toggle functionality for 1 schedule would be available for the time being.

**Reply by tsightler** [reply]
> And I would imagine most of your users have enough knowledge to manually delete an entity from HA if it came to that.

I want to live in this imaginary world!  :)

**Comment by KruseLuds** [comment]
You can simply include switches in your dashboard to enable or disable whatever funtionality you want, and in the code wherever that functionality is handled, simply add a check on whether or not the appropriate switch is enabled or disabled (and such items can then be ignored if disabled). I have started adding this kind of functionality to my dashbopards for enabling and disabling a variery of processes. Although this has in the beginning of my automation journey started to make things increasingly complicated, I have also reduced that issue by replacing sets of redundant code I had to reuse in different automations - with calls in each case to a generic script (with in some cases a large number of parameters). This has made the entire situation much more elegant (and easier to maintain). Make sure the scripts are always set to be able to run in parallel with a large limit of instances. In some cases I now have dozens of automations all calling one script for a bunch of the logic.  Hope that helps!

**Reply by JB09** [reply]
I appreciate your reply, but without going on a huge tangent the issue is that there is no service an HA entity can call (via automation or helper-script) to accomplish this action at the moment as the functionality needs to be added to ring-mqtt first.  This is the reason for this discussion.
